### PR TITLE
#193 by peec: Initial implementation of paper-dialog.

### PR DESCRIPTION
**Pull request by DanChadwick**
To-Do list for paper-dialog

- [x] Rebase of peed/feature/dialog on wip/v1.0.0.
- [x] Update to current coding and documentation standards, including camelCase of actions.
- [x] Rename ember-paper-container to ember-paper-parent
- [ ] Fix transitions. The transition is not removed in the dummy app custom dialog when the dialog is closed.
- [ ] Replace transition builtin with https://github.com/peec/ember-css-transitions if possible.
- [ ] Replace keyword alert and confirm with title and infer confirm-style alert from presence of cancel.
- [ ] Focus on ok button. Provision to disable for custom dialog that have input elements.
- [ ] Handle key events: Return on focused button should click the button. Escape should close.
- [ ] Allow ember-paper-parent to be omitted, if possible. I had difficulty putting ember-paper-parent high up in the DOM hierarchy, which I believe should be possible. For example, place in an ancestor route's template.
- [ ] Allow dialogOrigin to not be specified. Default to $('body').
- [ ] Cleanup by @miguelcobain "too much jquery".
- [ ] Remove param from actions in dummy app. Use closures instead.
- [ ] More tests.

This is a rebase of peec/feature/dialog on wip/v1.0.0. I resolved the (few) conflicts (using angular-source, a little suave style) and fixed a couple of tiny bugs (including including sidenav-theme -- unrelated, but I fixed it when I saw it). There may be more work to do, but at least maybe we can get peec's work into the wip/v1.0.0 branch where it can receive further attention.

### CODE REVIEW

File: addon/components/paper-dialog-container.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
```
Comment (miguelcobain) [inline addon/components/paper-dialog-container.js]: use `const { Component, computed } = Ember;` style to conform new commits

File: addon/components/paper-dialog-content.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import PaperDialogInner from './paper-dialog-inner';
+
+export default Ember.Component.extend({
+  tagName: 'md-dialog-content',
+  classNames: ['md-dialog-content'],
+
+  dialogInnerComponent: Ember.computed(function () {
+    return this.nearestOfType(PaperDialogInner);
+  }),
+  imagesLoaded () {
```
Comment (miguelcobain) [inline addon/components/paper-dialog-content.js]: this should be triggering jscs errors.

Run `jscs -c .jscsrc app addon` to see style errors.

File: addon/mixins/translate3d-mixin.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
+import Ember from 'ember';
+
+export default Ember.Mixin.create({
```
Comment (miguelcobain) [inline addon/mixins/translate3d-mixin.js]: Can this mixin be replaced by https://github.com/peec/ember-css-transitions, @peec?

**Comment by DanChadwick** [comment]
Failure (finally) unrelated to this PR.

**Comment by miguelcobain** [comment]
API proposal:
```hbs
{{#if modalOpen}}
  {{#paper-dialog onClose=(action "toggleModal") as |dialog|}}
    {{#dialog.content as |content|}}
      {{#content.title}} Title Here {{/content.title}}
      {{#content.body}}
        A longer description here. A longer description here.
      {{/content.body}}
    {{/dialog.content}}
    {{#dialog.actions as |actions|}}
      {{#actions.close-button}}
        Close this thing
      {{/actions.close-button}}
      {{#my-custom-thing onClick=(action actions.closeAction)}}
        Close it in another way!
      {{/my-custom-thing}}
    {{/dialog.actions}}
  {{/paper-dialog}}
{{/if}}
```

- rendering `{{#paper-dialog` must show the dialog
- Following the DDAU principle, the way to close the dialog would be for it to send an action and let the user app toggle the `{{#if`
- it should use ember-wormhole (this PR probably already does that?
- animations should work
- also, tests

File: tests/dummy/app/templates/dialog.hbs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+
+    {{#if confirmDialogIsOpen}}
+      {{paper-dialog
+      confirm="Would you like to delete your debt?"
+      content="All of the banks have agreed to forgive you your debts."
+      ok="Please do it!"
+      cancel="Sounds like a scam"
+      on-ok="confirmClearDept"
+      on-cancel="cancelClearDept"
```
Comment (miguelcobain) [inline tests/dummy/app/templates/dialog.hbs]: Let's use camelCase action names and closure actions.

**Comment by DanChadwick** [comment]
- Closing animation is broken

**Comment by DanChadwick** [comment]
Closed in preference to work in miguelcobain:dialogs-jj-abrams branch.

**Pull request closed without merging**